### PR TITLE
Added radix-sort

### LINKED ISSUE #813: Add radix sort in sorting Algorithms
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Description

 Add radix sort in sorting Algorithms

### Suggested Change

It should include radix sort which is needed.

### Rationale

_No response_

### Urgency

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Hemav009**
## Related Issue

Fix #813 

## Description

Added radix-sort

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

File: docs/dsa/arrays/radix-sort.md [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    public static void main(String[] args) {
+        int[] arr = {170, 45, 75, 90, 802, 24, 2, 66};
+        radixSort(arr);
+        System.out.println(Arrays.toString(arr));
+    }
+}
+```
+
+### Complexity of Radix Sort
+
```
Comment (Ajay-Dhangar) [inline docs/dsa/arrays/radix-sort.md:252]: ### Bad Practice

```md

- **Time Complexity**: O(d⋅(n+k))
  - $d: Number of digits in the largest number.$
  - $n: Number of elements in the array.
  - $k: Range of the digits for decimal system,k = 10.$
- **Space Complexity**: O(n + k)
```

### Good Practice

```md
- **Time Complexity**: $O(d⋅(n+k))$
  - $d$: Number of digits in the largest number.
  - $n$: Number of elements in the array.
  - $k$: Range of the digits for decimal system, $k = 10.$
- **Space Complexity**: $O(n + k)$
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Hemav009** [comment]
@Ajay-Dhangar  updated

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**